Port 2.x 30-actions.sh to 3.0

ArangoGutierrez opened the pull request:
**Description of the Pull Request (PR):**

Port 2.x 30-actions.sh to 3.0 

**Checkoff for all PRs:**

- [x] I have read the [Guidelines for Contributing](https://github.com/sylabs/singularity/blob/master/CONTRIBUTING.md), and this PR conforms to the stated requirements.
- [ ] I have added changes to the [CHANGELOG](https://github.com/sylabs/singularity/blob/development/CHANGELOG.md) and and documentation updates to the [user](https://www.github.com/sylabs/singularity-userdocs) and [administrator](https://www.github.com/sylabs/singularity-admindocs) documentation bases.
- [x] I have tested this PR locally with a `make testall`
- [x] This PR is against the project's master branch
- [x] I have added myself as a contributor to the [contributors's file](https://github.com/sylabs/singularity/blob/master/CONTRIBUTORS.md)
- [x] This PR is ready for review and/or merge


Attn: @singularity-maintainers

ArangoGutierrez committed:
Add singularity run tests

Signed-off-by: ArangoGutierrez <eduardo@sylabs.io>

ArangoGutierrez committed:
Add test suit for actions cmd

Port 30-actions.sh script to 3.0

Signed-off-by: ArangoGutierrez <eduardo@sylabs.io>

ArangoGutierrez committed:
Fix copy/paste typos on test names

Signed-off-by: ArangoGutierrez <eduardo@sylabs.io>

jscook2345 reviewed (commented):
So some of this is duplicated in other tests. The actions test itself is non super specific. I would prefer breaking this out into each action. There already exists an 'exec_test'. Perhaps create a 'shell_test', and a 'run_test' to contain those parts.

bauerm97 commented on src/cmd/singularity/actions_test.go:
```diff
@@ -0,0 +1,185 @@
+	pwd     string
+}
+
+func imageExec(opts execOpts, imagePath string, command []string) ([]byte, error) {
```
What is the purpose of this function if it never gets called?

ArangoGutierrez replied:
that func is used by `build_test.go` and `docker_test.go` , not in this file. as we need to catch the exit code

bauerm97 commented on src/cmd/singularity/actions_test.go:
```diff
@@ -0,0 +1,185 @@
+
+	for _, tt := range tests {
+		t.Run(tt.name, test.WithoutPrivilege(func(t *testing.T) {
+			cmd := exec.Command(tt.binName, tt.argv...)
```
If we've created a function to run an image, then why are we not using it?

ArangoGutierrez replied:
`imageExec()` does not catch and return the exit code. should we modify that func to do it. or re allocate `imageExec()` to a diff test file

ArangoGutierrez committed:
Merge branch 'master' of https://github.com/singularityware/singularity into test_porting

ArangoGutierrez committed:
Update imageExec() for general test usage

func imageExec() func should be generic enough to run the singularity command for any test which needs to do so.
the updated func imageExec() can be used to run/exec/shell a Singularity image
it return the exitCode and err of the execution
func imageExec(t *testing.T, action string, opts execOpts, imagePath string, command []string)
(stdout string, stderr string, exitCode int, err error)

Signed-off-by: ArangoGutierrez <eduardo@sylabs.io>

ArangoGutierrez committed:
Update tests calling func imageExec()

Signed-off-by: ArangoGutierrez <eduardo@sylabs.io>

ArangoGutierrez committed:
Merge branch 'master' of https://github.com/singularityware/singularity into test_porting

bauerm97 committed:
Merge branch 'master' into test_porting
